akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -31,6 +32,31 @@ object Sink {
     new Sink(scaladsl.Sink.fold[U, In](zero)(f.apply).toCompletionStage())
 
   /**
+   * A `Sink` that will use Java 8 `Collector` to reduce and optionally transform
+   * accumulated result into a final representation.
+   * As collector is mutable you can use it only with one stream.
+   * Reduction operations performs sequentially.
+   */
+  def javaCollector[T, A, R](collector: java.util.stream.Collector[T, A, R]): Sink[T, CompletionStage[R]] =
+    new Sink(scaladsl.Sink.javaCollector[T, A, R](collector).toCompletionStage())
```

Comment:
@patriknw, after thinking a little. I would left javaCollector in Sink as it's not converting to some java stream it's do transformation inside sink so it's not a converter in pure sense.
However I'll move toJavaStream methods and as JavaStream (for Sink) as they converting from and to Streams.